compiler/rustc_feature/src/unstable.rs

```diff
@@ -483,6 +483,8 @@ declare_features! (
     (incomplete, generic_const_exprs, "1.56.0", Some(76560)),
     /// Allows generic parameters and where-clauses on free & associated const items.
     (incomplete, generic_const_items, "1.73.0", Some(113521)),
+    /// Allows usage of the generic `NonZero<T>` type.
+    (unstable, generic_nonzero, "1.77.0", Some(120257)),
```

Comment:
`CURRENT_RUSTC_VERSION`?

Reply:
I don't think this applies here, given it's not in a `stable` attribute.

Reply:
Wait, isn't `generic_nonzero` library feature?

Reply:
Yes, but this is the definition of that feature.

Reply:
I believe this list is only for language and compiler features. Library features are defined just by the use of `#[unstable(feature = "...")]` in the standard library.

Do you recall if there was a particular test failure that required this to be added here?

Reply:
> Do you recall if there was a particular test failure that required this to be added here?

All of them showed something like

```
unknown feature `generic_nonzero`
```

Reply:
Just checked again, it's only the `tests/ui/abi/compatibility.rs` test:

```
error[E0635]: unknown feature `generic_nonzero`
```

Reply:
Ah I see, it's because that test uses `no_core` on some platforms.

Reply:
Ah, I can reproduce the failure if I run `x.py test` against the parent commit of https://github.com/rust-lang/rust/commit/6b3c695f00b262acfcc3ad5865a61380674c95f0.

```console
---- [ui] tests/ui/abi/compatibility.rs#mips64 stdout ----

error in revision `mips64`: test compilation failed although it shouldn't!
status: exit status: 1
command: RUSTC_ICE="0" "/git/rust/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "/git/rust/tests/ui/abi/compatibility.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/home/david/.cargo" "--sysroot" "/git/rust/build/x86_64-unknown-linux-gnu/stage1" "--cfg" "mips64" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/git/rust/build/x86_64-unknown-linux-gnu/test/ui/abi/compatibility.mips64" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/git/rust/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/git/rust/build/x86_64-unknown-linux-gnu/test/ui/abi/compatibility.mips64/auxiliary" "--target" "mips64-unknown-linux-gnuabi64"
stdout: none
--- stderr -------------------------------
error[E0635]: unknown feature `generic_nonzero`
  --> /git/rust/tests/ui/abi/compatibility.rs:66:12
   |
LL | #![feature(generic_nonzero, rustc_attrs, unsized_fn_params, transparent_unions)]
   |            ^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0635`.
```

The issue is that in this particular test, if `cfg(not(host))`, then it builds with `no_core` and thus `core::num::NonZero` would never be defined.

https://github.com/rust-lang/rust/blob/6b3c695f00b262acfcc3ad5865a61380674c95f0/tests/ui/abi/compatibility.rs#L67

I think https://github.com/rust-lang/rust/commit/37b3922b1e0b3d4cc1e59fd30adedd286bc6b99f is the right way to fix this. `#![cfg_attr(host, feature(generic_nonzero))]`

Reply:
Thanks, I was just about to push the exact same fix. 😄